bindings/python/edlib.pyx

```
@@ -152,3 +154,91 @@ def align(query, target, mode="NW", task="distance", k=-1, additionalEqualities=
+    ## target_pos = locations[0][0]
+    target_pos = alignResult["locations"][0][0]
+    if target_pos == None:
+        target_pos = 0
+    query_pos = 0 ## 0-indexed
+    target_aln = match_aln = query_aln = ""
+
+    cigar = alignResult["cigar"]
+
+    ###
+    ## cigar parsing, motivated by yech1990: https://github.com/Martinsos/edlib/issues/127
+    ##
+    ## 'num_occurences' == Number of occurrences of the alignment operation
+    ## 'alignment_operation' == Cigar symbol/code that represent an alignment operation
+    ## 
+    for num_occurrences, alignment_operation in re.findall("(\d+)(\D)", cigar):
+        num_occurrences = int(num_occurrences)
+        if alignment_operation == "=":
+            target_aln += target[target_pos : target_pos + num_occurrences]
+            target_pos += num_occurrences
+            query_aln += query[query_pos : query_pos + num_occurrences]
+            query_pos += num_occurrences
+            match_aln += "|" * num_occurrences
+        elif alignment_operation == "X":
+            target_aln += target[target_pos : target_pos + num_occurrences]
+            target_pos += num_occurrences
+            query_aln += query[query_pos : query_pos + num_occurrences]
+            query_pos += num_occurrences
+            match_aln += "." * num_occurrences
+        elif alignment_operation == "D":
+            target_aln += target[target_pos : target_pos + num_occurrences]
+            target_pos += num_occurrences
+            query_aln += gapSymbol * num_occurrences
+            query_pos += 0
+            match_aln +=gapSymbol * num_occurrences
+        elif alignment_operation == "I":
+            target_aln += gapSymbol * num_occurrences
+            target_pos += 0
+            query_aln += query[query_pos : query_pos + num_occurrences]
+            query_pos += num_occurrences
+            match_aln += gapSymbol * num_occurrences
+        else:
+            pass
```

Can it be something else? This should never happen, right? If so, instead of passing, let's throw. Ohterwise we could hide errors, or produce result for faulty cigar and user will never know.

---

I don't think it can be anything else, but it's possible there are odd cigar's out there in fields I don't know about. 

I'll throw an error here instead of passing. If there are issues, users should open a github PR.